### PR TITLE
Favoriting a multi-configuration job breaks blueocean favourites

### PR DESCRIPTION
**Pull request by sophistifunk**
# Description

See [JENKINS-47835](https://issues.jenkins-ci.org/browse/JENKINS-47835).

# Submitter checklist
- [X] Link to JIRA ticket in description, if appropriate.
- [X] Change is code complete and matches issue description
- [X] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by michaelneale** [comment]
very neat code @sophistifunk looks good

code review @imeredith @vivek ?

File: blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/util/Capabilities.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     *
+     * @param item a Jenkins Item / BO object
+     * @param capability the string name of the capability you wish to search for
+     * @return true if annotation is found
+     */
+    public static boolean hasCapability(Object item, String capability) {
+        return item != null && capability != null && Capabilities.hasCapability(item.getClass(), capability);
+    }
+
+    private static boolean hasCapability(@Nonnull Class<?> clazz, @Nonnull String capability) {
```
Comment (imeredith) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/util/Capabilities.java]: Imo this should be public. Its oossible that the caller only has the class instance and not the object (although unlikely)

**Comment by michaelneale** [comment]
the failure doesn't seem related to this change. It is to do with editing and parallel: 



(screen shot from the artifacts tab)

running it again...

File: blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/util/Capabilities.java (hunk 1 above)
Comment (vivek) [reply blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/util/Capabilities.java]: +1

**Comment by michaelneale** [comment]
:bee:

**Pull request merged**